src/python/pants/build_graph/target.py

```diff
@@ -794,12 +760,44 @@ def strict_dependencies(self, dep_context):
     """
     strict_deps = self._cached_strict_dependencies_map.get(dep_context, None)
     if strict_deps is None:
+      closure_kwargs = dep_context.target_closure_kwargs
+      respect_intransitive = closure_kwargs.get('respect_intransitive')
+      include_scopes = closure_kwargs.get('include_scopes')
+      exclude_scopes = closure_kwargs.get('exclude_scopes')
+      def pred(target, dep):
+        if type(target) in dep_context.alias_types:
+          return True
+        if respect_intransitive and not dep.transitive:
+          return False
+        if not dep.scope.in_scope(
+            include_scopes=include_scopes,
+            exclude_scopes=exclude_scopes):
+          return False
+
+        if target is self: # then just include all the direct deps.
```

Comment:
Hm. This case is odd... do we do something like passing all roots as `dep_predicate(t, t)` ?

Reply:
dep_predicate is only called on prior to walking a dependency edge. `self` is the root of the traversal, so it is always walked. So, if the source param of `dep_predicate` is `self`, then we must be looking at a direct dep of `self`. 


Another approach, now that I think of it, would be to traverse from `self.dependencies`. Then we wouldn't need a `self` check. On the other hand, then we'd need to check the scopes of those dependencies outside the traversal.